app/services/update_account_service.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class UpdateAccountService < BaseService
+  def call(account, params)
+    before_locked = account.locked
+    account.update(params).tap do |ret|
+      next unless ret
+      AuthorizeAllFollowsWorker.perform_async(account.id) if before_locked && !account.locked
+      # authorize_all_follow_requests(account) if before_locked && !account.locked
+    end
```

Comment:
I think it is fine to do synchronously because almost all operations in `AuthorizeFollowService` are done by workers.